nanovdb/nanovdb/NanoVDB.h

```diff
@@ -529,92 +742,93 @@ __hostdev__ inline static const T* alignPtr(const T* p)
+// --------------------------> PtrDiff <------------------------------------
 
-template <typename T1, typename T2>
+/// @brief Compute the distance, in bytes, between two pointers
+/// @tparam T1 Type of the first pointer
+/// @tparam T2 Type of the second pointer
+/// @param p fist pointer, assumed to NOT be NULL
+/// @param q second pointer, assumed to NOT be NULL
+/// @return signed distance between pointer addresses in units of bytes
+template<typename T1, typename T2>
 __hostdev__ inline static int64_t PtrDiff(const T1* p, const T2* q)
 {
     NANOVDB_ASSERT(p && q);
     return reinterpret_cast<const char*>(p) - reinterpret_cast<const char*>(q);
 }
 
-template <typename DstT, typename SrcT>
-__hostdev__ inline static DstT* PtrAdd(SrcT *p, int64_t offset)
+// --------------------------> PtrAdd <------------------------------------
+
+/// @brief Adds a byte offset of a non-const pointer to produce another non-const pointer
+/// @tparam DstT Type of the return pointer
+/// @tparam SrcT Type of the input pointer
+/// @param p non-const input pointer, assumed to NOT be NULL
+/// @param offset signed byte offset
+/// @return a non-const pointer defined as the offset of an input pointer
+template<typename DstT, typename SrcT>
+__hostdev__ inline static DstT* PtrAdd(SrcT* p, int64_t offset)
 {
     NANOVDB_ASSERT(p);
     return reinterpret_cast<DstT*>(reinterpret_cast<char*>(p) + offset);
 }
 
-template <typename DstT, typename SrcT>
-__hostdev__ inline static const DstT* PtrAdd(const SrcT *p, int64_t offset)
+/// @brief Adds a byte offset of a const pointer to produce another const pointer
+/// @tparam DstT Type of the return pointer
+/// @tparam SrcT Type of the input pointer
+/// @param p const input pointer, assumed to NOT be NULL
+/// @param offset signed byte offset
+/// @return a const pointer defined as the offset of a const input pointer
+template<typename DstT, typename SrcT>
+__hostdev__ inline static const DstT* PtrAdd(const SrcT* p, int64_t offset)
 {
     NANOVDB_ASSERT(p);
     return reinterpret_cast<const DstT*>(reinterpret_cast<const char*>(p) + offset);
 }
 
-// --------------------------> Rgba8 <------------------------------------
+// --------------------------> isFloatingPoint(GridType) <------------------------------------
 
-/// @brief 8-bit red, green, blue, alpha packed into 32 bit unsigned int
-class Rgba8
+/// @brief return true if the GridType maps to a floating point type
+__hostdev__ inline bool isFloatingPoint(GridType gridType)
 {
-    union {
-        uint8_t  c[4];// 4 color channels of red, green, blue and alpha components.
-        uint32_t packed;// 32 bit packed representation
-    } mData;
-public:
-    static const int SIZE = 4;
-    using ValueType = uint8_t;
+    return gridType == GridType::Float ||
+           gridType == GridType::Double ||
+           gridType == GridType::Fp4 ||
+           gridType == GridType::Fp8 ||
+           gridType == GridType::Fp16 ||
+           gridType == GridType::FpN;
```

Comment:
What about `GridType::Half`?

Reply:
Good catch - I'll add it. However, GridType::Half a little strange since in practice it is not (yet) used anywhere. The reason is I   introduce GridType::Half for IEEE 16 half floating points (as implemented in OpenEXR), but in reality NanoVDB is only using GridType::Fp16 which is my own (quantized) implementation of 16-bit floating points. As soon as OpenVDB supports Half in memory grids (we currently only support it out-of-core) then NanoVDB should also support it, in which case nanovdb::isFloatingPoint(GridType::Half) should also work. However, there is no harm in fixing this now :) So, consider this fixed ...